tables.go

```diff
@@ -42,81 +44,79 @@ func (lx *LXRHash) Log(msg string) {
 // HashSize is the number of bits in the hash; truncated to a byte bountry
 // Passes is the number of shuffles of the ByteMap performed.  Each pass shuffles all byte values in the map
 func (lx *LXRHash) Init(Seed, MapSizeBits, HashSize, Passes uint64) {
+	// These panics are here to maintain the interface (using
+	// panic instead of returning an error) while reusing the
+	// code in InitFromPath.
 	if MapSizeBits < 8 {
 		panic(fmt.Sprintf("Bad Map Size in Bits.  Must be between 8 and 34 bits, was %d", MapSizeBits))
 	}
+	TablePath, err := GetUserTablePath()
+	if err != nil {
+		panic(err)
+	}
+	// lx.readTableFromPath does not make directories.
+	err = os.MkdirAll(TablePath, os.ModePerm)
```

Comment:
It is here because this is where the current/legacy interface is implemented.  I specifically tried to keep directory creation out of the path of the new interface.  The idea is that the existence of the directory is the responsibility of the code which provides the path for the table file.

Again, if you want the library to be mucking about with directories, I'll put it back, but it feels like creating directories does not belong in a hashing library.  (If we follow this thinking to its full conclusion, perhaps we should be using `InitWithFileHandle` instead of `InitWithPath`.)

Reply:
Right now you have one os.MkDirAll in the Init() function, one os.MkDirAll in the ReadTable function (which is now no longer used), and none in the InitFromPath. 

Init() first creates the folder, then immediately checks to see if the folder exists, which is redundant. It makes more sense for Init() to do nothing and InitFromPath to check if the folder exists, then for the write function to be the one that creates the folder. The functionality would be unchanged, but the creation of the folder code is where it's supposed to be, next to the code that writes the file, and the function that wants an extra safeguard has the extra safeguard code in it.

I'm not entirely sold on the justification, though:

> The idea is that the existence of the directory is the responsibility of the code which provides the path for the table file.
> ... but it feels like creating directories does not belong in a hashing library.

Other hashing libraries don't do it because other hashing libraries don't create large files on disk. LXRHash needs that file to function well, so it has code to manage that aspect. 

That said it's just a matter of preference. I'd be okay with it working this way if everything else checks out.

Reply:
The MkdirAll has no effect for InitFromPath because the check has already thrown the error if the directory does not exist.  The old code still gets the MkdirAll.  This works for me.  The changes were not trivial, so you may want to take a look at 7dd1fca